### PR TITLE
Fix tests and command line tool

### PR DESCRIPTION
**Pull request by zaolin**
See #142

### CODE REVIEW

File: pkg/test/memory.go [hunk 1]
```diff
@@ -124,13 +124,6 @@ var (
 		SpecificiationTitle:     IntelTXTSpecificationTitle,
 		SpecificationDocumentID: IntelTXTSpecificationDocumentID,
 	}
-	testnosiniterrors = Test{
-		Name:        "SINIT ACM startup successful",
-		Required:    false,
-		NonCritical: true,
-		function:    NoSINITErrors,
-		Status:      Implemented,
-	}
```
Comment (ChriMarMe) [inline pkg/test/memory.go:133]: Why was this removed?

File: pkg/test/test.go [hunk 2]
```diff
@@ -54,15 +54,23 @@ const (
...
+const (
+	// FWCoreboot indicates test does run on coreboot firmware
+	FWCoreboot FirmwareType = "coreboot"
+
+	// FWNoCoreboot indicates test doesn't run on coreboot firmware
+	FWNoCoreboot FirmwareType = "UEFI"
+)
+
 func (t Result) String() string {
-	return [...]string{"TESTNOTRUN", "DEPENDENCY_FAILED", "INTERNAL_ERROR", "FAIL", "WARN", "PASS"}[t]
```
Comment (ChriMarMe) [inline pkg/test/test.go:65]: Why remove the concept of warn?

File: pkg/test/tpm.go [hunk 3]
```diff
@@ -895,39 +846,26 @@ func NPWModeIsNotSetInPS(txtAPI hwapi.APIInterfaces) (bool, error, error) {
...
+			return true, fmt.Errorf("Auto-promotion mode active"), nil
 		}
 	}
 	if pol2 != nil {
+		if pol2.PolicyType != tools.LCPPolicyTypeAny {
+			return true, fmt.Errorf("Signed Policy mode active"), nil
+		}
 		if pol2.PolicyType != tools.LCPPolicyTypeList {
-			return false, fmt.Errorf("Auto-promotion mode active"), nil
+			return true, fmt.Errorf("Auto-promotion mode active"), nil
```
Comment (ChriMarMe) [inline pkg/test/tpm.go:868]: Why we dont keep the granularity?

File: pkg/test/memory.go (hunk 1 above)
Comment (ChriMarMe) [reply pkg/test/memory.go:133]: Also seperate commit

File: pkg/test/test.go (hunk 2 above)
Comment (ChriMarMe) [reply pkg/test/test.go:65]: Also seperate commit.

File: pkg/test/tpm.go (hunk 3 above)
Comment (ChriMarMe) [reply pkg/test/tpm.go:868]: Maybe you guessed it ;) Separate commit

File: pkg/test/cpu.go [hunk 4]
```diff
@@ -287,7 +286,7 @@ func IBBIsTrusted(txtAPI hwapi.APIInterfaces) (bool, error, error) {
 	if regs.BootStatus&(1<<59) != 0 && regs.BootStatus&(1<<63) != 0 {
 		return true, nil, nil
 	}
-	return false, fmt.Errorf("IBB not trusted"), err
+	return true, fmt.Errorf("IBB not trusted, MLE not running"), err
```
Comment (PatrickRudolph) [inline pkg/test/cpu.go:289]: That's impossible. If something returns true all error must be nil.

File: pkg/test/acpi.go [hunk 5]
```diff
@@ -19,6 +19,7 @@ var (
 		Required:                true,
 		function:                CheckRSDPValid,
 		Status:                  Implemented,
+		Firmware:                FWNoCoreboot,
```
Comment (PatrickRudolph) [inline pkg/test/acpi.go:22]: The new Field Firmware is just wrong, whatever it does. That's not clear from the commit message.

File: pkg/tools/smbios.go [hunk 6]
```diff
@@ -0,0 +1,27 @@
+package tools
+
+import (
+	"errors"
+
+	"github.com/digitalocean/go-smbios/smbios"
+)
+
+// SMBIOSGetVendor gets the vendor name from table 0
+func SMBIOSGetVendor() (*string, error) {
```
Comment (PatrickRudolph) [inline pkg/tools/smbios.go:10]: that must be astracted over the hwapi.
